doc/UsersGuide/source/packagemanager.rst

```diff
@@ -18,17 +18,35 @@ project package and a certain version of a read-only package (or library), which
 and that you drag-and-drop components from.
 
 This dependency is automatically marked in your package by adding a `uses annotation
-<https://specification.modelica.org/maint/3.5/annotations.html#version-handling>`_ at the top level. For example, if you
+<https://specification.modelica.org/maint/3.6/annotations.html#version-handling>`_ at the top level. For example, if you
 drag and drop components from MSL 4.0.0 into models of your package, the ``annotation(uses(Modelica(version="4.0.0")));``
 will be added automatically to it. This information allows OpenModelica to automatically load all the libraries
 that are required to compile the models in your own package next time you (or someone else, possibly on a different
 computer) loads your package, provided they are installed in places on the computer's file system where OpenModelica
 can find them.
 
 The default place where OpenModelica looks for packages is the so-called
-`MODELICAPATH <https://specification.modelica.org/maint/3.5/packages.html#the-modelica-library-path-modelicapath>`_.
+`MODELICAPATH <https://specification.modelica.org/maint/3.6/packages.html#the-modelica-library-path-modelicapath>`_.
 You can check where it is by typing ``getModelicaPath()`` in the Interactive Environment (Tools | OpenModelica Compiler CLI in OMEdit).
-Installed read-only libraries are placed by default in the MODELICAPATH.
+Installed read-only libraries are all placed by default in the MODELICAPATH.
+
+However, when you open a package from the file system, before looking into MODELICAPATH, OpenModelica will first look for
+packages it depends upon in the same directory that contains the package you just opened. For example, if you open
+``/home/John/ModelicaPackages/MoonShot/package.mo``, and your MoonShot package contains ``annotation(uses(Rockets));``,
+before looking into MODELICAPATH, OpenModelica will try to load ``/home/John/ModelicaPackages/Rockets/package.mo`` or
+``/home/John/ModelicaPackages/Rockets.mo``, if the package is stored in a single file. So, if you are developing
+several packages with dependencies among them, you can place them in the same common root directory to make sure
+that all the dependencies are loaded automatically. 
```

Comment:
This is kind of what normally happens, but it's not entirely correct. We append the directory the package is in to the MODELICAPATH and load them as normal libraries, which means that we check all of the directories in the MODELICAPATH for the best version to load before actually trying to load anything.

The result is usually the same as what you describe, but with the subtle difference that if a "better" version is found elsewhere in the MODELICAPATH then the compiler will load that one instead. In other words, we do not prioritize libraries based on their location in the filesystem.

Reply:
I see, but wouldn't it be better to do it how it is described there, i.e., prepend the directory the package is in to the MODELICAPATH, instead of appending it?

I mean, if I want to work on development versions of some otherwise read-only installed libraries, the former should have the precedence to the latter. What would be the rationale of leaving the directory the package is in as a last resort?

Reply:
It is actually prepended, but the order doesn't really matter since we search all of the directories for the best version to load.

Reply:
[MLS 13.3](https://specification.modelica.org/maint/3.6/packages.html#the-modelica-library-path-modelicapath) says that MODELICAPATH is an _ordered list_, but then it never explains what that order means.

I'll fix the User's Guide according to the current behaviour.

Reply:
It also says that a tool may use the version information to search for a specific version of a library, which is kind of what we're doing.

If a version isn't given it seems we load the last library found in the MODELICAPATH, i.e. if `MODELICAPATH=dir1:dir2` and both contain `lib` with no version then we load `lib` from `dir2`. This is perhaps a bit surprising, but given that no one has ever complained about as far as I know I guess it doesn't really matter in practice. And as you say the specification doesn't really say what the order is.